tests/cupy_tests/polynomial_tests/test_polynomial.py

```diff
@@ -0,0 +1,57 @@
+import unittest
+
+import numpy
+import pytest
+
+import cupy
+from cupy import testing
+
+
+@testing.gpu
+class TestPolynomial(unittest.TestCase):
```

Comment:
Could you add more edge cases?
- The input array has a non-zero value, but `ValueError` should be raised (e.g., `xp.array([3, 0, 0, 0, 0, 0])`).
- The input has one non-zero value, but `ValueError` is not raised (e.g., `xp.array([0, 0, -2, 0, 0, 0])`).